### PR TITLE
chore(monocdk): generate fewer files

### PR DESCRIPTION
**Pull request by RomainMuller**
In order to reduce the risk of conflicting with a "legitimate" source
file from a module, stop writing an `index.ts` file in the module's
relocated home, and instead refer to the modules' entry points directly.
This makes the rewriter part of the packaging ever so slightly more
complicated, but actually reduces the amount of indirection layers that
will be traversed during a `require` (run-time) or `import` (build-time).

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/monocdk-experiment/build-tools/gen.ts [hunk 1]
```diff
@@ -228,7 +228,7 @@ async function transformPackage(
 
     await fs.writeFile(
       path.resolve(LIB_ROOT, '..', `${library.shortName}.ts`),
-      `export * from './lib/${library.shortName}';\n`,
+      `import { ${library.namespace} } from './lib';\nexport = ${library.namespace};\n`,
```
Comment (MrArnoldPalmer) [inline packages/monocdk-experiment/build-tools/gen.ts:231]: Just curious, why this change? I'm not super familiar with the `export = ` syntax but according to the TS docs
>When exporting a module using export =, TypeScript-specific import module = require("module") must be used to import the module.

https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

Comment (RomainMuller) [reply packages/monocdk-experiment/build-tools/gen.ts:231]: I basically wanted to get rid of the `index.ts` that was previously introduced *only* to back this particular entry point. But I reckon it is possible to achieve the same by using `export { namespace } from './lib';`.
